R-package/configure

```diff
@@ -1828,7 +1822,18 @@ main ()
 
 
 _ACEOF
-    ${CC} -o conftest conftest.c ${OPENMP_LIB} ${OPENMP_CXXFLAGS} 2>/dev/null && ./conftest && ac_pkg_openmp=yes
+    ${CXX} ${CPPFLAGS} ${CXXFLAGS} ${LDFLAGS} ${OPENMP_CXXFLAGS} ${OPENMP_LIB} -o conftest conftest.cpp 2>/dev/null && ./conftest && ac_pkg_openmp=yes
+
+    # -Xclang is not portable (it is clang-specific)
+    # if compilation above failed, try without that flag
```

Comment:
@jameslamb Sorry for commenting in closed PR.
I just thought about this a little bit.
This if branch is mostly for "not AppleClang compiler", right? Then why do we test only `-fopenmp` flag? Maybe it's worth checking some other flags, e.g. `-qopenmp` and `-openmp` for Intel compiler?
Possible values: https://github.com/autoconf-archive/autoconf-archive/blob/6c25681d78d963708362ea0d4d63ee9b508782b6/m4/ax_openmp.m4#L82

https://www.gnu.org/software/autoconf-archive/ax_openmp.html

Reply:
oooo interesting! Yes, this branch is for "not clang", and was inspired by what I saw other `data.table` doing (since I know those maintainers have spent a lot of time struggling with OpenMP detection in R packages).

For example, 

https://github.com/Rdatatable/data.table/blob/4cf92893742a0359ce3b3ddfd279a9e69627d52b/configure#L91

```shell
# Compilation failed -- try forcing -fopenmp instead.
R_NO_OPENMP=0
"${CC}" "${CFLAGS}" -fopenmp test-omp.c || R_NO_OPENMP=1
```

I didn't know about these other flags that you've mentioned, but seems like a good idea to try for even more portability! If you open an issue listing those other flags (with the links you've included here) and and which compilers they are recognized by, I'd be happy to work on adding and testing that support.

Reply:
> If you open an issue ...

Sure, just created #4537.